### ISSUE TITLE
Report leaks with LeakSanitizer/AddressSanitizer before the program exits

### ISSUE THREAD

**Issue by manuel-arguelles** [google/sanitizers#1386]
Hi,

It seems that memory leaks are only reported when the program exists, is it possible to report them earlier? or what's the correct way for evaluating possible leaks in program that are meant to behave like a server/daemon (with an endless event loop)?

I tried sending different signals to interrupt the process, but in those cases nothing gets reported.

**Comment by LazyHippogriff** [comment]
I do the following to get leak reports during runtime. Basically you need to use a function from lsan_interface.h. I put this inside a signal handler to get the reports only when I send a particular signal to my program.

`
#include <lsan_interface.h>
#include <thread>
#include <chrono>
#include <signal.h>

using namespace std::literals::chrono_literals;


void signalHandlerSigCont(int fa_sigNum){

      fprintf(stdout,"[%s::%d]--> PFCP proxy received SIGCONT\n",__FILE__,__LINE__);

      //Output Present time to stdout
      time_t my_time = time(NULL);
      fprintf(stdout,"PFCP proxy received SIGCONT at time:%s\n", ctime(&my_time));

    __lsan_do_recoverable_leak_check();

  }

void threadFunc(int* fa_obj) {
  //delete [] fa_obj;

}

void threadGenerator(int* fa_obj) {
  std::thread aThread = std::thread(threadFunc,fa_obj);
  aThread.detach();
}

int main() {
     signal(SIGCONT,signalHandlerSigCont);

  while(true) {
    int *a = new int[5];
    threadGenerator(a);
    std::this_thread::sleep_for(50ms);
  }
}

`

**Comment by manuel-arguelles** [comment]
Thanks @LazyHippogriff
